src/components/views/rooms/RoomSettings.js

```diff
@@ -45,81 +31,165 @@ module.exports = React.createClass({
     },
 
     getInitialState: function() {
-        // work out the initial color index
-        var room_color_index = undefined;
-        var color_scheme_event = this.props.room.getAccountData("org.matrix.room.color_scheme");
-        if (color_scheme_event) {
-            var color_scheme = color_scheme_event.getContent();
-            if (color_scheme.primary_color) color_scheme.primary_color = color_scheme.primary_color.toLowerCase();
-            if (color_scheme.secondary_color) color_scheme.secondary_color = color_scheme.secondary_color.toLowerCase();
-            // XXX: we should validate these values
-            for (var i = 0; i < room_colors.length; i++) {
-                var room_color = room_colors[i];
-                if (room_color[0] === color_scheme.primary_color &&
-                    room_color[1] === color_scheme.secondary_color)
-                {
-                    room_color_index = i;
-                    break;
-                }
-            }
-            if (room_color_index === undefined) {
-                // append the unrecognised colours to our palette
-                room_color_index = room_colors.length;
-                room_colors[room_color_index] = [ color_scheme.primary_color, color_scheme.secondary_color ];
-            }
-        }
-        else {
-            room_color_index = 0;
-        }
-
         var tags = {};
         Object.keys(this.props.room.tags).forEach(function(tagName) {
             tags[tagName] = {};
         });
 
+        var are_notifications_muted = false;
+        var roomPushRule = MatrixClientPeg.get().getRoomPushRule("global", this.props.room.roomId); 
+        if (roomPushRule) {
```

Comment:
State vars are mildly useful to keep as underscores because they map directly to the state key being modified.